spinalcordtoolbox/scripts/sct_deepseg.py

```diff
@@ -208,15 +213,26 @@ def main(argv: Sequence[str]):
     for name_model in name_models:
         # Check if this is an official model
         if name_model in list(models.MODELS.keys()):
+            # Check if the custom path is provided
+            if arguments.custom_path is not None:
+                path_model = arguments.custom_path
```

Comment:
Because `-custom-path` requires users to download/extract/memorize the path of the new model, I'm wondering if it would be easier for the user if we instead consider one of these options:

1. Use `-custom-url` instead (so that if we provide the user with the asset URL, they can download it using `sct_deepseg -install` instead). I'm imagining the usage would be something like:
    ```bash
    sct_deepseg -install seg_spinal_rootlets_t2w -custom-url https://github.com/ivadomed/model-spinal-rootlets/releases/download/r20240523/model-spinal-rootlets_ventral_D106_r20240523.zip
    ```
    I think this would be pretty easy, since all of the install logic is already present. We would just have to make a small change to the `install_model()` function in `models.py` so that it accepts a `custom_url` parameter, and that we set `url_field = [custom_url] if custom_url else MODELS[name_model]['url']`. This will bypass the original URL and use the custom URL instead.
2. Alternatively, a solution that requires *zero* modifications would be to just provide them with a `curl`/`wget` + `unzip` commands that will automatically extract the `.zip` to the correct folder. For your new rootlets model, this can be done using the commands:
    ```bash
    export TASK_NAME="seg_spinal_rootlets_t2w"
    export MODEL_PATH="$SCT_DIR/data/deepseg_models/model_seg_spinal_rootlets_nnunet"
    export NEW_MODEL_URL="https://github.com/ivadomed/model-spinal-rootlets/releases/download/r20240523/model-spinal-rootlets_ventral_D106_r20240523.zip"
    sct_deepseg -install "$TASK_NAME"           # Generate the initial folder + source.json
    rm -rf "$MODEL_PATH"/nnUNetTrainer*         # Remove the existing model folder
    curl -L -o /tmp/model.zip "$NEW_MODEL_URL"  # Download model `.zip`
    unzip /tmp/rootlets.zip -d "$MODEL_PATH"    # Replace with updated model version
    ```

However, `sct_deepseg` would fail because the new `.zip` does not contain an `nnUNetTrainer` subfolder, hehe:

![image](https://github.com/user-attachments/assets/7fdba9ca-ac66-44f0-8392-28c05e3bf373)

Reply:
>Hmm. The argument -help description says "used to specify a custom model that is not part of the official list of tasks." (i.e. not available with -task)

> However, in the code, the logic for if arguments.custom_path is not None comes inside the block if name_model in list(models.MODELS.keys()). Doesn't that mean that the user would have to specify an official -task in order to make this work?

Good catch! You are absolutely right. The help is supposed to be: "used to specify a custom model that is part of the official list of tasks." (i.e. removing "not")

Reply:
> Because -custom-path requires users to download/extract/memorize the path of the new model ...

Good point. I was wondering how problematic this could be for users. Your idea with `-custom-url` (below) is probably better (and easier for users).

> Use -custom-url instead (so that if we provide the user with the asset URL, they can download it using sct_deepseg -install instead). 

This sounds like a good idea! The only thing I find slightly unintuitive is that the asset URL is different from the release URL (as described in the "URL support" section in my [first comment](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4618#issue-2498909018)). But as you mentioned, if we provide the users with the asset URL, everything will be fine.

Reply:
That's a good point. I can see the benefit to providing the URL to the release (because it contains useful information about the release).

Maybe what we could do is provide instructions like this:

> To quickly test a new release of a model (before it is officially added to SCT):
> 1. Navigate to the [Releases page](https://github.com/ivadomed/model-spinal-rootlets/releases) for the model.
> 2. Open the page for the release you would like to test (e.g. [r20240523](https://github.com/ivadomed/model-spinal-rootlets/releases/tag/r20240523) - ventral rootlets).
> 3. Scroll down to the "**Assets**" section.
> 4. Find the `model.zip` asset, right click it, then copy the link. 
> 5. Install the model by running `sct_deepseg -install seg_spinal_rootlets_t2w -custom-url https://www.github.com/model.zip` (Paste the link that you copied.)
> 6. Run the model as normal by running `sct_deepseg -task seg_spinal_rootlets_t2w -i t2.nii.gz`

That way, they get the useful information as part of the release, but still know that they need to copy the asset link.

Reply:
@valosekj I can work on the above refactor to `-custom-url` if you would like? (That way, you are free to work on the lesion length/width draft PR.)

Reply:
This would be great! Thank you, @joshuacwnewton!
just FYI: I'm now updating the [r20240523](https://github.com/ivadomed/model-spinal-rootlets/releases/tag/r20240523) ventral rootlets model ([details](https://github.com/ivadomed/model-spinal-rootlets/issues/77)) to be compatible with `sct_deepseg`